elastic-agent-client.proto

```diff
@@ -177,6 +178,78 @@ enum UnitType {
     OUTPUT = 1;
 }
 
+// Log level for the unit.
+enum UnitLogLevel {
+    TRACE = 0;
+    DEBUG = 1;
+    INFO = 2;
+    WARN = 3;
+    ERROR = 4;
+}
+
+// Package metadata provided in the meta field of a unit.
+message Package {
+    // Source is the original source of the message. All values from the message are included here
+    // even if other concrete fields are defined for this message.
+    google.protobuf.Struct source = 1;
+    // Name of the package.
+    string name = 2;
+    // Version of the package.
+    string version = 3;
+}
+
+// Metadata provided in the meta field of a unit.
+message Meta {
+    // Source is the original source of the message. All values from the message are included here
+    // even if other concrete fields are defined for this message.
+    google.protobuf.Struct source = 1;
+    // Package metadata.
+    Package package = 2;
+}
+
+// Data stream defined in either top-level unit or in multiple streams in the unit.
+message DataStream {
+    // Source is the original source of the message. All values from the message are included here
+    // even if other concrete fields are defined for this message.
+    google.protobuf.Struct source = 1;
+    // Dataset for the stream.
+    string dataset = 2;
+    // Type for the stream.
+    string type = 3;
+    // Namespace for the stream.
+    string namespace = 4;
+}
+
+// Stream defined in a configuration.
+message Stream {
+    // Source is the original source of the message. All values from the message are included here
+    // even if other concrete fields are defined for this message.
+    google.protobuf.Struct source = 1;
+    string id = 2;
+    DataStream data_stream = 3;
```

Comment:
Is it worth including a repeated list of processors? Likely the type would have to be `google.protobuf.Struct` as I don't think we want to encode the beats processor structure here, so it probably just saves us a key looking in `source`.

Reply:
If we want to define a proper message for `processors` then I would be okay with placing it in here. I believe the policy should be its either a proper message or you look for it in `source`. There is no real benefit unless its a proper message.

Reply:
Yah, the processors are inconsistent enough that we'd have to define all of them, which would be an...annoying process. I'm leaning towards just not defining them, since I don't think a `[]string` is super useful. We can add it later if needed, though.

Reply:
I agree with @fearful-symmetry there are indeed all different here, I don't think we need it initially so I would be more relax for it.